onset_maproom/calc.py

```diff
@@ -33,6 +35,123 @@ def onset_date(daily_rain, early_start_day, early_start_month, search_days, rain
+  """
+  #Deal with leap year cases
+  if start_day == 29 and start_month == 2 :
+    start_day = 1
+    start_month = 3
+  end_day2 = end_day
+  end_month2 = end_month
+  if end_day == 29 and end_month == 2 :
+    end_day2 = 1
+    end_month2 = 3
+  #Drop date outside very first and very last edges -- this ensures we get complete seasons with regards to edges, later on
+  start_edges = daily_data[time_coord].where(
+    ((daily_data[time_coord].dt.day==start_day) & (daily_data[time_coord].dt.month==start_month)),
+    drop=True
+  )
+  end_edges = daily_data[time_coord].where(
+    ((daily_data[time_coord].dt.day==end_day2) & (daily_data[time_coord].dt.month==end_month2)),
+    drop=True
+  )
+  daily_data = daily_data.sel(**{time_coord: slice(start_edges[0],end_edges[-1])})
+  start_edges = daily_data[time_coord].where(
+    ((daily_data[time_coord].dt.day==start_day) & (daily_data[time_coord].dt.month==start_month)),
+    drop=True
+  )
+  end_edges = daily_data[time_coord].where(
+    ((daily_data[time_coord].dt.day==end_day2) & (daily_data[time_coord].dt.month==end_month2)),
+    drop=True
+  )
+  #Creates array of edges of the season that will form the bins
+  seasons_edges = xr.concat([start_edges, end_edges], "T_out", join="override")
+  #Creates seasons_starts that will be used for grouping
+  #and seasons_ends that is one of the outputs
+  if end_day == 29 and end_month == 2 :
+    days_in_season = (
+      daily_data[time_coord] >= seasons_edges.isel(T_out=0).rename({time_coord: "group"})
+    ) & (
+      daily_data[time_coord] < seasons_edges.isel(T_out=1).rename({time_coord: "group"})
+    )
+  else:
+    days_in_season = (
+      daily_data[time_coord] >= seasons_edges.isel(T_out=0).rename({time_coord: "group"})
+    ) & (
+      daily_data[time_coord] <= seasons_edges.isel(T_out=1).rename({time_coord: "group"})
+    )
+  seasons_starts = daily_data[time_coord].where(days_in_season)
+  seasons_starts = (
+    xr.where(seasons_starts >= seasons_starts["group"], seasons_starts["group"], seasons_starts)
+    .min(dim="group", skipna=True)
+    .dropna(dim=time_coord)
+    .rename("seasons_starts")
+  )
+  seasons_ends = (
+    daily_data[time_coord].where(days_in_season)
+    .max(dim="T", skipna=True)
+    .rename("seasons_ends")
+  )
+  #Drops daily data not in seasons of interest
+  daily_data = daily_data.where(seasons_starts, drop=True)
+  #Group by season with groups labeled by seasons_starts
+  daily_tobegroupedby_season = xr.merge([daily_data, seasons_starts, seasons_ends])
+  return daily_tobegroupedby_season
+
+def seasonal_sum(daily_data, start_day, start_month, end_day, end_month, min_count=None, time_coord="T"):
+  """Calculates seasonal totals of daily data in season defined by day-month edges
+  """
+  #It turns out that having daily_groupby_season concatenate only every other group is not enough to drop entired the undesired seasons
+  #sum will return NaN for these so need to use dropna to clean up
```

Comment:
Is this comment out of date? I don't see the dropna it refers to.

Reply:
is now removed